src/store/api/index.ts

```diff
@@ -0,0 +1,88 @@
+import { Category, Good, GoodInBasket } from "../../types";
+
+const BASE_URL = "http://localhost:3000";
+
+const getData = (url: string, params = {}) => {
+  const searchParams = new URLSearchParams({ ...params });
+  const fullUrl = new URL(url, BASE_URL);
+
+  fullUrl.search = searchParams.toString();
+
+  return fetch(fullUrl).then((data) => {
+    if (data.ok) {
+      return data.json();
+    }
+    throw new Error("oops");
+  });
+};
+
+let token = "";
+const postData = (url: string, body: any) => {
+  const fullUrl = new URL(url, BASE_URL);
+  return fetch(fullUrl, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(body),
+  })
+    .then((data) => {
+      if (data.ok) {
+        return data.json();
+      }
+    })
+    .then((response) => {
+      token = response.token;
+      return response;
+    });
+};
+
+const postBasket = (url: string, body: Record<string, unknown>) => {
+  return fetch(new URL(url, BASE_URL), {
+    method: "PUT",
+    body: JSON.stringify(body),
+    headers: { "Content-Type": "application/json" },
```

Comment:
Во все запросы (post / get / put / delete) добавь токен, чтобы авторизация заработала
```suggestion
    headers: { "Content-Type": "application/json", Authorization: ` Bearer ${token}` },
```